src/test/java/org/ambraproject/rhino/service/impl/VolumeCrudServiceImplTest.java

```diff
@@ -0,0 +1,185 @@
+import org.mockito.stubbing.Answer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.hibernate3.HibernateTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration(classes = VolumeCrudServiceImplTest.class)
+@Configuration
+public class VolumeCrudServiceImplTest extends AbstractStubbingArticleTest {
+
+  private VolumeCrudService mockVolumeCrudService;
+
+  private JournalCrudService mockJournalCrudService;
+
+  private VolumeOutputView.Factory mockVolumeOutputViewFactory;
+
+  private Query mockQuery;
+
+  private HibernateTemplate mockHibernateTemplate;
+
+  private Volume stubVolume = createStubVolume();
+
+  private Journal stubJournal = createStubJournal();
+
+  private final String VOLUME_CREATION_JSON = "{\"doi\": \"10.1371/volume.pmed.v01\", \"displayName\": \"2004\"}";
+
+  private VolumeInputView stubVolumeInputView;
+
+  @Bean
+  public VolumeCrudService volumeCrudService() {
+    mockVolumeCrudService = spy(VolumeCrudServiceImpl.class);
+    LOG.debug("volumeCrudService() * --> {}", mockVolumeCrudService);
+    return mockVolumeCrudService;
+  }
+
+  @Bean
+  public VolumeOutputView.Factory volumeOutputViewFactory() {
+    mockVolumeOutputViewFactory = mock(VolumeOutputView.Factory.class);
+    LOG.debug("volumeOutputViewFactory() * --> {}", mockVolumeOutputViewFactory);
+    return mockVolumeOutputViewFactory;
+  }
+
+  private Volume createStubVolume() {
+    final Volume stubVolume = new Volume("test");
+    stubVolume.setIssues(new ArrayList<>());
+    stubVolume.setLastModified(new Date());
+    return stubVolume;
+  }
+
+  private Journal createStubJournal() {
+    final Journal stubJournal = new Journal("test");
+    stubJournal.setVolumes(new ArrayList<>());
+    return stubJournal;
+  }
+
+  /**
+   * Returns a mock query used to mock a HibernateTemplate.
+   */
+  private Answer<Query> createQueryAnswer() {
+    return invocation -> {
+      final String sql = invocation.getArgument(0);
+
+      // Return appropriate data based on SQL.
+      LOG.info("sql: {}", sql);
+      if (sql.contains("FROM Volume")) {
+        when(mockQuery.uniqueResult()).thenReturn(stubVolume);
+      } else if (sql.contains("FROM Journal")) {
+        when(mockQuery.uniqueResult()).thenReturn(stubJournal);
+      }
+      return mockQuery;
+    };
+  }
+
+  /**
+   * Returns a mock query used to mock a HibernateTemplate that returns no values.
+   */
+  private Answer<Query> createEmptyQueryAnswer() {
```

Comment:
This took me a minute to figure out. Mockito callback methods, such as `thenReturn()`, do not allow null values such as `null` or `Optional.empty()`.

`createVolume()` below has logic that will throw an exception if _any_ volume is found. This means that we have to somehow return an empty `Optional` object from the `mockHibernateTemplate`.

This 'empty' `Answer` was the simplest way I could think of to get the empty result I wanted. Definitely open to a more succinct way of doing this if we find one.

Reply:
:+1: